### PR TITLE
feat(pie-monorepo): DSW-2172 pie-design-tokens update

### PR DESCRIPTION
**Pull request by dandel10n**
## Describe your changes (can list changeset entries if preferable)

---
"wc-vanilla": minor
"pie-storybook": minor
"pie-docs": minor
"pie-monorepo": minor
---

[Changed] - pie-design-tokens version bump to 6.5.0.

---
"@justeattakeaway/pie-css": minor
---

[Changed] - replaces hardcoded breakpoint values with PIE tokens.

---
"@justeattakeaway/pie-cookie-banner": patch
---

[Removed] - pie-design-tokens dependency.

---
"@justeattakeaway/pie-notification": minor
"@justeattakeaway/pie-modal": minor
---

[Changed] - replaces media with include-media with pie breakpoints tokens.

It also updates the usage of scss var for ::backdrop in modal component with css value since the bug has been fixed. I still had to leave the old scss option for Safari <v17.6 temporary until we drop the support of v16. 
- https://wpt.fyi/results/html/semantics/interactive-elements/the-dialog-element/backdrop-inherits.html?label=experimental&label=master&aligned
- https://issues.chromium.org/issues/40569411
- https://bugzilla.mozilla.org/show_bug.cgi?id=1855668
- https://bugs.webkit.org/show_bug.cgi?id=263834

I checked in Browserstack the following browsers:
Firefox: 130, 129, 128
Chrome: 128, 127, 126
Opera: 113, 112, 111
Edge: 128, 127, 126
Safari: 17.6 worked fine on my machine but didn't show the backdrop in browserstack on older versions.

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have used the `/snapit` functionality to test my changes in a consuming application

### Reviewer 1 @kevinrodrigues 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2 @maledr5 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-cookie-banner/package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
   "license": "Apache-2.0",
   "devDependencies": {
     "@custom-elements-manifest/analyzer": "0.9.0",
-    "@justeat/pie-design-tokens": "6.3.1",
+    "@justeat/pie-design-tokens": "6.5.0",
```
Comment (raoufswe) [inline packages/components/pie-cookie-banner/package.json:40]: We don't need to list the dependency here as it's indirectly imported from `@justeattakeaway/pie-css`. 

@xander-marjoram can you double-check that, please?

File: packages/components/pie-modal/package.json [hunk 2]
```diff
@@ -36,7 +36,7 @@
   "license": "Apache-2.0",
   "devDependencies": {
     "@custom-elements-manifest/analyzer": "0.9.0",
-    "@justeat/pie-design-tokens": "6.3.1",
+    "@justeat/pie-design-tokens": "6.5.0",
```
Comment (raoufswe) [inline packages/components/pie-modal/package.json:39]: same here

Comment (dandel10n) [reply packages/components/pie-modal/package.json:39]: For pie-modal it is needed for styling overlay, see [here](https://github.com/justeattakeaway/pie/blob/main/packages/components/pie-modal/src/modal.scss#L123)

Comment (dandel10n) [reply packages/components/pie-modal/package.json]: I removed the usage and removed the dependency

**Review by maledr5 — APPROVED** [review]

**Review by kevinrodrigues — APPROVED** [review]

**Pull request merged**